contracts/src/utils/cryptography/ecdsa.rs

```diff
@@ -0,0 +1,248 @@
+//! Elliptic Curve Digital Signature Algorithm (ECDSA) operations.
+//!
+//! These functions can be used to verify that a message was signed
+//! by the holder of the private keys of a given address.
+use alloc::vec::Vec;
+
+use alloy_primitives::{address, uint, Address, B256, U256};
+use alloy_sol_types::{sol, SolType};
+use stylus_proc::SolidityError;
+use stylus_sdk::{
+    call::{self, Call},
+    storage::TopLevelStorage,
+};
+
+const ECRECOVER_ADDR: Address =
+    address!("0000000000000000000000000000000000000001");
+
+const EIP2_VALUE: U256 = uint!(
```

Comment:
This should be more concrete about what it refers to because the name is a bit ambiguous. Maybe `SIGNATURE_S_UPPER_BOUND`? Also, probably should be public.